### PR TITLE
Finished 'Builtin and Custom Objects' homework #2

### PR DESCRIPTION
**Pull request by daktest**
- wrote a Television class with notes on how to use it

[#151556921]

### CODE REVIEW

File: objects/television.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+=end
+
+class Television
+
+  CHANNEL_RANGE = 3..500
+  VOLUME_RANGE = 0..25
+
+  def initialize
+    @is_on = false
```
Comment (wb97531) [inline objects/television.rb]: conventionally speaking, a '?' should be used for when a method name or instance variable returns a boolean

**Comment by wb97531** [comment]
well done, to clean it up a bit, look for some opportunities to use some attr_readers

**Comment by daktest** [comment]
I don't see a use for attr_reader here.  I had originally used :channel and :volume but that allowed a user to retrieve the channel and volume when the Television was not turned on which was behaviour I didn't want.

**Comment by wb97531** [comment]
does 'on?' mean something different then 'powered', why are two methods necessary?

so for example: you could just have an attr_reader for on? and get rid of @powered and on? method

**Comment by daktest** [comment]
There aren't two methods; @powered is just a variable; on? is a method.

How do you create an attr_reader with a '?' in it?  I tried and got an error.  Am I missing something?

File: objects/television.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env ruby
```
Comment (jasonnoble) [inline objects/television.rb]: No space between `#!` and `/usr/bin...`

File: objects/television.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    @pre_mute_volume = 0
+  end
+
+  def power
+    @on = !@on
+    puts "TV is now #{on? ? 'on' : 'off'}"
+  end
+
+  def set_channel(number)
+    if on?
```
Comment (jasonnoble) [inline objects/television.rb]: You may want to change this to a logic gate...

    def set_channel(number)
      return unless on?
      ...
   end

File: objects/television.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  def volume_down
+    if on?
+      @volume -= 1 if @volume > VOLUME_RANGE.first
+      volume_status
+    end
+  end
+
+  def mute
+    if on?
+      if @volume == 0
```
Comment (jasonnoble) [inline objects/television.rb]: `if muted`

**Review by jasonnoble — CHANGES_REQUESTED** [review]

File: objects/television.rb (hunk 2 above)
Comment (daktest) [reply objects/television.rb]: Gah, I keep doing that!  Don't know why, I don't do it in Perl!

**Pull request merged**